packages/smart-router/evm/fot.ts

```diff
@@ -0,0 +1,86 @@
+import { ChainId } from '@pancakeswap/chains'
+import { WNATIVE } from '@pancakeswap/sdk'
+import { Address, PublicClient, getContract } from 'viem'
+import { feeOnTransferDetectorAbi } from './abis/FeeOnTransferDetector'
+
+export const feeOnTransferDetectorAddresses = {
+  [ChainId.ZKSYNC]: '0xED87D01674199355CEfC05648d17E037306d7962',
+  [ChainId.ZKSYNC_TESTNET]: '0x869505373d830104130F95c1E7d578dE7E58C0a8',
+} as const
+
+const getFeeOnTransferDetectorContract = <TPublicClient extends PublicClient>(publicClient: TPublicClient) => {
+  if (publicClient.chain && publicClient.chain.id in feeOnTransferDetectorAddresses) {
+    return getContract({
+      abi: feeOnTransferDetectorAbi,
+      address: feeOnTransferDetectorAddresses[publicClient.chain.id as keyof typeof feeOnTransferDetectorAddresses],
+      publicClient,
+    })
+  }
+
+  return null
+}
+
+const AMOUNT = 100000n
```

Comment:
Is this raw amount? Shall we decide the amount based on the token decimal?

Reply:
It is the minimum amount for bps. Use the smallest one to prevent slippage affecting the call

Reply:
ok